supplier/views.py

```diff
@@ -10,6 +13,34 @@ def get_supplier(user):
         return False
 
 
+def receive_product(request, supplier):
+    prod_name = request.GET['searched_product']
+    if prod_name:
+        product = Product.filter_name(prod_name).first()
+        supplier_product = SupplierProduct.objects.filter(qr_code=product,
+                                                          user_name=supplier)
+        if supplier_product.exists():
+            return supplier_product
+    return None
+
+
+def action_search_product(request, supplier_product):
+    if supplier_product:
+        return render(request, 'supplier/suppliers.html', {'supplier_products': supplier_product})
+    else:
+        messages.info(request, 'Product Not Found')
+        return render(request, 'supplier/suppliers.html')
+
+
+def action_display_all_products(request, supplier):
+    data = set(SupplierProduct.objects.filter(user_name=supplier))
+    return render(request, 'supplier/suppliers.html', {'supplier_products': set(data)})
```

Comment:
```suggestion
    return render(request, 'supplier/suppliers.html', {'supplier_products': data})
```

Reply:
i had to make it a set, query sets are non comparable
` data = SupplierProduct.objects.filter(user_name=supplier)
assert data == SupplierProduct.objects.filter(user_name=supplier)`
will result as fail.

Reply:
had to do it for the test, query sets are non comparable.
` data = set(SupplierProduct.objects.filter(user_name=supplier))
assert data == set(SupplierProduct.objects.filter(user_name=supplier) `
will result as fail.
and since i used mock.call_args_list, i couldnt change it inside the list, so when it was called it was already a set.

Reply:
Line 36 has `data = set(something)` and then line 37 makes this set a set once more. Looks redundant.

Reply:
yea, didnt notice, will remove👍